x-pack/plugins/security_solution/public/app/home/global_header/index.tsx

```diff
@@ -37,63 +38,75 @@ const BUTTON_ADD_DATA = i18n.translate('xpack.securitySolution.globalHeader.butt
  * This component uses the reverse portal to add the Add Data, ML job settings, and AI Assistant buttons on the
  * right hand side of the Kibana global header
  */
-export const GlobalHeader = React.memo(
-  ({ setHeaderActionMenu }: { setHeaderActionMenu: AppMountParameters['setHeaderActionMenu'] }) => {
-    const portalNode = useMemo(() => createHtmlPortalNode(), []);
-    const { theme } = useKibana().services;
-    const { pathname } = useLocation();
+export const GlobalHeader = React.memo(() => {
+  const portalNode = useMemo(() => createHtmlPortalNode(), []);
+  const { theme, setHeaderActionMenu, i18n: kibanaServiceI18n } = useKibana().services;
+  const { pathname } = useLocation();
 
-    const getTimeline = useMemo(() => timelineSelectors.getTimelineByIdSelector(), []);
-    const showTimeline = useShallowEqualSelector(
-      (state) => (getTimeline(state, TimelineId.active) ?? timelineDefaults).show
-    );
+  const getTimeline = useMemo(() => timelineSelectors.getTimelineByIdSelector(), []);
+  const showTimeline = useShallowEqualSelector(
+    (state) => (getTimeline(state, TimelineId.active) ?? timelineDefaults).show
+  );
 
-    const sourcererScope = getScopeFromPath(pathname);
-    const showSourcerer = showSourcererByPath(pathname);
+  const sourcererScope = getScopeFromPath(pathname);
+  const showSourcerer = showSourcererByPath(pathname);
+  const dashboardViewPath =
+    matchPath(pathname, {
+      path: `/${SecurityPageName.dashboards}/:id`,
+      exact: false,
+      strict: false,
+    }) != null;
 
-    const { href, onClick } = useAddIntegrationsUrl();
+  const { href, onClick } = useAddIntegrationsUrl();
 
-    useEffect(() => {
-      setHeaderActionMenu((element) => {
-        const mount = toMountPoint(<OutPortal node={portalNode} />, { theme$: theme.theme$ });
-        return mount(element);
+  useEffect(() => {
+    setHeaderActionMenu((element) => {
+      const mount = toMountPoint(<OutPortal node={portalNode} />, {
+        theme,
+        i18n: kibanaServiceI18n,
       });
+      return mount(element);
+    });
 
-      return () => {
-        portalNode.unmount();
-        setHeaderActionMenu(undefined);
-      };
-    }, [portalNode, setHeaderActionMenu, theme.theme$]);
-
-    return (
-      <InPortal node={portalNode}>
-        <EuiHeaderSection side="right">
-          {isDetectionsPath(pathname) && (
-            <EuiHeaderSectionItem>
-              <MlPopover />
-            </EuiHeaderSectionItem>
-          )}
+    return () => {
+      /* Dashboard mounts an edit toolbar, it should be restored when leaving dashboard editing page */
```

Comment:
To fix Global action missing:


https://github.com/elastic/kibana/assets/6295984/46202af4-ec95-4a1b-a8fd-db38897da3d7



After:

https://github.com/elastic/kibana/assets/6295984/94f04248-fc6c-4129-90d5-91fa2b09b244